### ISSUE TITLE
NPE during upload with 1.113.12

### ISSUE THREAD

**Issue by n12c** [googleapis/java-storage#745]
com.google.cloud:libraries-bom:19.0.0 (com.google.cloud:google-cloud-storage:jar:1.113.12), we sometimes see:

```
com.google.cloud.storage.StorageException: java.lang.NullPointerException
	at com.google.cloud.storage.StorageException.translateAndThrow(StorageException.java:81)
	at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:209)
	at com.google.cloud.BaseWriteChannel.close(BaseWriteChannel.java:151)
	at com.google.cloud.storage.StorageImpl.createFrom(StorageImpl.java:265)
    ...snip...
Caused by: java.lang.NullPointerException: null
	at com.google.cloud.storage.spi.v1.StorageRpc$Option.get(StorageRpc.java:85)
	at com.google.cloud.storage.spi.v1.StorageRpc$Option.getString(StorageRpc.java:89)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.setEncryptionHeaders(HttpStorageRpc.java:242)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.getCall(HttpStorageRpc.java:439)
	at com.google.cloud.storage.spi.v1.HttpStorageRpc.get(HttpStorageRpc.java:455)
	at com.google.cloud.storage.BlobWriteChannel.getRemoteStorageObject(BlobWriteChannel.java:80)
	at com.google.cloud.storage.BlobWriteChannel.access$500(BlobWriteChannel.java:33)
	at com.google.cloud.storage.BlobWriteChannel$1.run(BlobWriteChannel.java:174)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.storage.BlobWriteChannel.flushBuffer(BlobWriteChannel.java:154)
	... 31 common frames omitted
```

Same call succeeds when retried.

**Comment by frankyn** [comment]
Thanks @n12c 

Does your usage look similar to:

```java
 BlobId blobId = BlobId.of(bucketName, key);
 BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
 storage.createFrom(blobInfo, ...);
```

**Comment by n12c** [comment]
Indeed.

```
public long put( String name, InputStream stream, long length, SecretKey key ) {
    var blobInfo = BlobInfo.newBuilder( bucket, name ).build();
    var bufferSize = length == -1 ? uploadMaxBufferSize : (int) Math.min( length, uploadMaxBufferSize );
    var options = key == SentinelKey.PLAINTEXT ?
        new BlobWriteOption[]{ BlobWriteOption.disableGzipContent() } :
        new BlobWriteOption[]{ BlobWriteOption.disableGzipContent(), BlobWriteOption.encryptionKey( key ) };
    return Unchecked.supplier( () ->
        gcs.createFrom( blobInfo, stream, bufferSize, options ).getSize() ).get();
}
```

**Comment by frankyn** [comment]
Thanks @n12c, I'm working on PR to resolve this. I can't release today (Friday today). Earliest will be Monday. 

How often does this failure occur and how many objects are you uploading in your workflows?

**Comment by n12c** [comment]
That's fine, this code is not in production.

Not sure precisely how many objects, but I see a few thousand PUT and POST calls to GCS in my APM. I see the NPE in my logs two times, but I only adopted 1.113.12 a couple of days ago.

**Comment by frankyn** [comment]
Thanks for clarifying, could you clarify `APM`? 
Also do you know how big are your objects on average?

**Comment by n12c** [comment]
https://www.datadoghq.com/product/apm/

Objects sizes are bi-modal, one population is very small (hundreds of bytes) but these are not encrypted so probably not of interest here. The other population is encrypted and spans a wide range of sizes from kilobytes to hundreds of megabytes. Metrics explorer in the GCP console gives me a present bucket size of 799.486GiB and 16.565M objects (split evenly between populations) for an average size of ~47KiB across the whole bucket.

**Comment by frankyn** [comment]
Thanks @n12c, I cut a release with the fix in version 1.113.13. PLMK if this has addressed the issue for you. From my experiments locally, this resolved potential NPE from passing in a `null` to a Get request in the library.
